build-tools/android/build.gradle

```diff
@@ -10,7 +10,7 @@ gradlePlugin {
             id = 'aditogether.android.app'
             implementationClass = 'aditogether.buildtools.android.AndroidAppPlugin'
         }
-        compose {
+        composePlugin {
```

Comment:
All the `NamedDomainObjectContainer`, like this one, are translating the DSL key (e.g. `composePlugin`) to `register("composePlugin")`.
You don't need to change the name to avoid clashes, you can use:
```
register("compose") {
  ...
}
```
Instead of:
```
compose {
  ...
}
```

Reply:
I wanted to avoid a different pattern. I was actually thinking about renaming them all as `xPlugin`. What do you think?

Reply:
Fine for me! Also, if you want to align them, can you also remove `displayName` and `description` from some of the declarations since they are not useful unless we publish them on Maven?

Reply:
I pushed the change.

I also did another change: the IDE shows a warning for some dependency notations from Version Catalog: e.g. if we have `libs.something` and `libs.something.sub`, there is a warning when using `libs.something`. Not sure if that's expected or a bug, but it seems a legit scenario to me. Have you got more info about it?

Reply:
I've not any additional insights on IDE warnings, but in general, we should avoid having `libs.something.sub` and `libs.something` IMHO because it's very confusing.

Usually, in Gradle catalogs, every part between dots is considered a group.
Following this idea, if `libs.something` and `libs.something.sub` exist, I wouldn't understand what's the content of `libs.something`, should it contain `libs.something.sub` or not?

TL;DR don't know what this IDE warning is, but IMHO we shouldn't have `libs.something` and `libs.something.sub` anyway

Reply:
The IDE it's warning about something like the usage of a group, indeed. In this case `libs.gradle.kotlin`, because we also have `libs.gradle.kotlin.serialization`

Reply:
(anyway, I think we can also fix/change this in another PR in case)

Reply:
IMHO for the moment we can just remove the serialization plugin since we are not using it. In future we can think about this problem if we have other warnings. What do you think?

Reply:
Fair enough, done